platform/src/main/java/org/hillview/sketches/FreqKList.java

```diff
@@ -17,208 +17,133 @@
-     * Prunes the hashmap to retain only those RowSnapshots that occur with frequency above
-     * 1/maxSize, and their frequencies.
+     * Filters the hashmap to retain only those RowSnapshots that occur with frequency above
+     * a specified threshold.
      */
-    public void filter(double threshold) {
-        for (ObjectIterator<Object2IntMap.Entry<RowSnapshot>> it = this.hMap.object2IntEntrySet().fastIterator();
-             it.hasNext(); ) {
-            final Object2IntMap.Entry<RowSnapshot> entry = it.next();
-            if (entry.getIntValue() < (threshold))
-                it.remove();
-        }
-    }
-
-    public void filter(boolean isMG) {
-        double threshold = this.epsilon * this.totalRows;
-        if (isMG)
-            threshold -= this.getErrBound();
-        filter(threshold);
-    }
-
-    public void rescale() {
-        for (ObjectIterator<Object2IntMap.Entry<RowSnapshot>> it = this.hMap.object2IntEntrySet().fastIterator();
-             it.hasNext(); ) {
+    protected void fkFilter(double threshold) {
+        for (ObjectIterator<Object2IntMap.Entry<RowSnapshot>> it =
+             this.hMap.object2IntEntrySet().fastIterator(); it.hasNext(); ) {
             final Object2IntMap.Entry<RowSnapshot> entry = it.next();
-            if (entry.getIntValue() < (this.epsilon * this.maxSize /1.1))
-                it.remove();
-            else
-                this.hMap.put(entry.getKey(), (int) Math.ceil(entry
-                        .getIntValue() * this.totalRows / this.maxSize));
+            if (entry.getIntValue() < threshold) it.remove();
         }
     }
 
     /**
-     * Type indicates which HeavyHitter Sketch is running.
-     * Type 0: the Misra-Gries sketch.
-     * Type 1: the exact Frequency sketch.
-     * Type 2: the Sampling frequency sketch.
+     * Method used to filter the table and keep only those rows that match one of the heavy hitters.
+     * @param schema tells us which columns were used to compare with the heavy hitters.
+     * @return A table filter for only those rows that match one of the heavy hitters in Schema.
      */
-    public Pair<List<RowSnapshot>, List<Integer>> getTop(int type) {
-        return getTop(this.hMap.size(), type);
+    public RowSnapshotSet.SetTableFilterDescription getFilter(final Schema schema) {
+        RowSnapshotSet rss = new RowSnapshotSet(schema, this.hMap.keySet());
+        return new RowSnapshotSet.SetTableFilterDescription(rss);
     }
 
     /**
-     * @param size: Lets us specify how many of the top items to select from the FreqKList.
+     * This is a helper method that takes two FreqKLists (left and right) and returns a hashmap with
+     * the union of their entries. If an element occurs in both, the frequencies add. This is used
+     * for both the Misra-Gries sketch and the sampling sketch. The hashmap is post-processed
+     * differently by each of them.
+     * (FeqKListExact implements its own addition and does not use this.)
```

Comment:
In general you should refrain of mentioning subclasses when you write code for a superclass - because when the code changes these references may become obsolete. But comment rot is generally an unsolved problem.